gapis/api/vulkan/transform_command_splitter.go

```diff
@@ -0,0 +1,934 @@
+				extraArgs = append(extraArgs, NewVkCmdEndRenderPassArgsʳ(inputState.Arena))
+			}
+			extraArgs = append(extraArgs, &InsertionCommand{
+				embedBuffer,
+				append([]VkCommandBuffer{}, splitTransform.pendingCommandBuffers...),
+				append(id, uint64(i)),
+				queueSubmit,
+			})
+			// If we were inside a renderpass, then we have to get back
+			// into a renderpass
+			if splitTransform.thisRenderPass != NilVkCmdBeginRenderPassArgsʳ {
+				extraArgs = append(extraArgs,
+					NewVkCmdBeginRenderPassArgsʳ(
+						inputState.Arena, VkSubpassContents_VK_SUBPASS_CONTENTS_INLINE,
+						splitTransform.currentRenderPass[splitTransform.thisSubpass][1], splitTransform.thisRenderPass.Framebuffer(), splitTransform.thisRenderPass.RenderArea(), splitTransform.thisRenderPass.ClearValues(),
+						splitTransform.thisRenderPass.DeviceGroupBeginInfo()))
+			}
+		}
+		if !replaceCommand {
+			cleanup, cmd, err := AddCommand(ctx, cb, embedBuffer, inputState, inputState, args)
+			if err != nil {
+				panic(fmt.Errorf("Invalid command-buffer detected %+v", err))
+			}
+			if err := splitTransform.observeAndWriteCommand(cmd); err != nil {
+				log.E(ctx, "Failed during adding command : [%v]%v", cmd, err)
+				return VkCommandBuffer(0), err
+			}
+			cleanup()
+		}
+		for _, ea := range extraArgs {
+			if ins, ok := ea.(api.Cmd); ok {
+				if err := splitTransform.observeAndWriteCommand(ins); err != nil {
+					log.E(ctx, "Failed during inserting insertion command : %v", err)
+					return VkCommandBuffer(0), err
+				}
+			} else {
+				cleanup, cmd, err := AddCommand(ctx, cb, embedBuffer, inputState, inputState, ea)
+				if err != nil {
+					panic(fmt.Errorf("Invalid command-buffer detected %+v", err))
+				}
+				if err := splitTransform.observeAndWriteCommand(cmd); err != nil {
+					log.E(ctx, "Failed during adding command : [%v]%v", cmd, err)
+					return VkCommandBuffer(0), err
+				}
+				cleanup()
+			}
+		}
+	}
+
+	return embedBuffer, nil
+}
+
+func (splitTransform *commandSplitter) getStartedCommandBuffer(ctx context.Context, queueSubmit *VkQueueSubmit, inputState *api.GlobalState) (VkCommandBuffer, error) {
+	cb := CommandBuilder{Thread: queueSubmit.Thread(), Arena: inputState.Arena}
+	queue := GetState(inputState).Queues().Get(queueSubmit.Queue())
+
+	commandPoolID, err := splitTransform.getCommandPool(ctx, queueSubmit, inputState)
+	if err != nil {
+		log.E(ctx, "Failed during getting command pool : %v", err)
+		return VkCommandBuffer(0), err
+	}
+
+	commandBufferAllocateInfo := NewVkCommandBufferAllocateInfo(inputState.Arena,
+		VkStructureType_VK_STRUCTURE_TYPE_COMMAND_BUFFER_ALLOCATE_INFO, // sType
+		NewVoidᶜᵖ(memory.Nullptr),                                      // pNext
+		commandPoolID,                                                  // commandPool
+		VkCommandBufferLevel_VK_COMMAND_BUFFER_LEVEL_PRIMARY,           // level
+		1, // commandBufferCount
+	)
+	commandBufferID := VkCommandBuffer(newUnusedID(true, func(x uint64) bool {
```

Comment:
These read the state, Have you made sure to mutate up to this point in order to not get a clashing commandbufferID?